### ISSUE TITLE
Spanner - Using UUID as a PrimaryKey

### ISSUE THREAD

**Issue by saturnism** [spring-cloud/spring-cloud-gcp#579]
I have a POJO that uses `java.util.UUID` as a primary key type mapped to Spanner type String(36). I created the converters.  However, when finding it by ID:
```
repository.findById(myUUID);
```

It throws an error:
```
java.lang.IllegalArgumentException: Unsupported type [java.util.UUID] for argument: f1cf9d40-6e99-4d1f-a717-6db99e60bd82
	at com.google.cloud.spanner.Key$Builder.appendObject(Key.java:187)
	at com.google.cloud.spanner.Key.of(Key.java:80)
	at org.springframework.cloud.gcp.data.spanner.repository.support.SimpleSpannerRepository.doIfKey(SimpleSpannerRepository.java:157)
```

It appears that we need to also convert the object to spanner type prior to constructing the `Key`.

**Comment by ChengyuanZhao** [comment]
This is a really good point. I think we should address this after PR #500 is in. 

The reason is that in that PR I add some code for generating the mapping from given Java types to Spanner compatible types taking into account what custom converters are provided. Making the UUID and other Java PK types work utilizing the given custom converters would be the exact same logic, 
so I think this would be the perfect thing to fix right after #500 is in.

**Comment by ChengyuanZhao** [comment]
@saturnism 
This is supposed to work now after #594.

**Comment by balopat** [comment]
closing as it is resolved in #594
